lib/yaml_normalizer/ext/nested.rb

```diff
@@ -17,21 +17,21 @@ module Nested
       #   => {"a"=>{"b"=>{"c"=>1}}, "b"=>{"x"=>2, "y"=>{"ok"=>true}, "z"=>4}}
       # @return [Hash] tree-shaped Hash
       def nested
-        tree = Hash.new { |h, k| h[k] = Hash.new(&h.default_proc) }
+        tree = {}
         each { |key, val| nest_key(tree, key.to_s, val) }
-        tree.default_proc = nil
         tree
       end
 
       private
 
       def nest_key(hash, key, val)
-        if key.include?('.')
-          keys = key.split('.')
-          hash.dig(*keys[0..-2])[keys.fetch(-1)] = val
-        else
-          hash[key] = val
+        keys = key.split('.')
+        last_key = keys.pop
+        sub_hash = hash
+        keys.each do |key|
```

Comment:
[rubocop] Please don't shadow outer local variable https://travis-ci.org/Sage/yaml_normalizer/builds/611976319#L471-L478

Reply:
Ok, so I renamed the input key to `dotted_key`.